File: data/EXP01/golang_no_system_var/2369/2369.output_1.go
func validPartition(nums []int) bool {
    n := len(nums)
    if n < 3 {
        return false
    }
    
    prev, count := nums[0], 1
    
    for i := 1; i < n; i++ {
        if nums[i] == prev {
            count++
        } else {
            if count > 2 {
                return true
            }
            prev = nums[i]
            count = 1
        }
    }
    
    return count > 2
}

File: data/EXP01/python/2369/2369.output_1.py
# @lc app=leetcode id=2369 lang=python3
class Solution:
    def validPartition(self, nums: List[int]) -> bool:
        n = len(nums)
        
        for i in range(n):
            if self.isValid(nums, i):
                return True
        
        return False
    
    def isValid(self, nums: List[int], start: int) -> bool:
        n = len(nums)
        count = 1
        
        if (start + 1 < n) and (nums[start] == nums[start + 1]):
            return True
        
        if (start + 2 < n) and (nums[start] == nums[start + 1] == nums[start + 2]):
            return True
        
        if (start + 2 < n) and (nums[start] + 1 == nums[start + 1]) and (nums[start + 1] + 1 == nums[start + 2]):
            return True
        
        return False